### ISSUE TITLE
How to use system's equalizer with exoplayer?

### ISSUE THREAD

**Issue by vyagrah** [google/ExoPlayer#10215]
I am using system default audio equalizer in my app.

For that I checked out this issue:
https://github.com/google/ExoPlayer/issues/3058

The issue told to use `onAudioSessionId(int audioSessionId)` meathod for applying audio effects before starting any playback, but I think it is deprecated.

I checked out these 2 meathods also, which are from
`AnalyticsListener`
```
exoPlayer.addAnalyticsListener(new AnalyticsListener() {
            @Override
            public void onAudioSessionIdChanged(EventTime eventTime, int audioSessionId) {
//                AnalyticsListener.super.onAudioSessionIdChanged(eventTime, audioSessionId);
                Log.i(TAG, "onAudioSessionIdChanged analytics: " + audioSessionId);
            }
        });
```
and `Player.Listener`
```
exoPlayer.addListener(new Player.Listener() {
            @Override
            public void onAudioSessionIdChanged(int audioSessionId) {
//                Player.Listener.super.onAudioSessionIdChanged(audioSessionId);
                Log.i(TAG, "onAudioSessionIdChanged listner: " + audioSessionId);
            }
});
```

my app is playing fine but **these 2 meathods never get called.**

Here is my media session creation code:
```
        mediaSessionCompat = new MediaSessionCompat(getApplicationContext(), Constants.MEDIA_SESSION_TOKEN);
        playerNotificationManager.setMediaSessionToken(mediaSessionCompat.getSessionToken());
        mediaSessionCompat.setActive(true);
```
What is the proper way to integrate exoplayer and system's equilizer ?

**Comment by marcbaechinger** [comment]
Please refer to the [answer provided](https://github.com/google/ExoPlayer/issues/3058#issuecomment-1108272570) in #3058. It basically tells to `use getAudioSessionId immediately after player construction to query the audio session ID that's being used`.

I'm going to close this issue as a duplicate of #3058. Please re-open if you think this it's needed.
